### PR TITLE
Extension to store features starting with vm-config in qubesdb

### PR DESCRIPTION
**Pull request by marmarta**
Idea: to have a simple way of transmitting certain external (qubes-level) settings to inside a vm.

references https://github.com/QubesOS/qubes-issues/issues/8138

### CODE REVIEW

File: qubes/ext/vm_config.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import qubes.ext
+import qubes.config
+
+PREFIX = 'vm-config.'
+
+
+class VMConfig(qubes.ext.Extension):
+    """This extension export features with 'vm-config.' prefix to QubesDB in
+    /qubes-vm-config/ tree.
```
Comment (marmarek) [inline qubes/ext/vm_config.py:31]: I'd drop `qubes-` part, qubesdb paths have quite low length limit, so lets leave more space for the actual name.

File: qubes/ext/vm_config.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    """
+
+    @qubes.ext.handler('domain-qdb-create')
+    def on_domain_qdb_create(self, vm, event):
+        """Actually export features"""
+        # pylint: disable=unused-argument
+        for feature, value in vm.features.items():
+            if not feature.startswith(PREFIX):
+                continue
+            service = feature[len(PREFIX):]
```
Comment (marmarek) [inline qubes/ext/vm_config.py:41]: It isn't really a service

File: qubes/ext/vm_config.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @qubes.ext.handler('domain-feature-delete:*')
+    def on_domain_feature_delete(self, vm, event, feature):
+        """Update /qubes-vm-config/ QubesDB tree in runtime"""
+        # pylint: disable=unused-argument
+        if not vm.is_running():
+            return
+        if not feature.startswith(PREFIX):
+            return
+        service = feature[len(PREFIX):]
```
Comment (marmarek) [inline qubes/ext/vm_config.py:83]: Here too

**Review by marmarek — CHANGES_REQUESTED** [review]
You also need to add the extension to setup.py. And to the spec file.

**Comment by marmarek** [comment]
PipelineRetry

**Comment by marmarta** [comment]
PipelineRetryFailed

**Pull request merged**